### PR TITLE
Fixes error/warnings found by shellcheck for ./packaging/installer/netdata-updater.sh

### PR DESCRIPTION
**Pull request by prologic**
See #7195

**Depend on:** #7937 (_review that first_)

##### Summary

As per title

##### Component Name

area/packaging

##### Additional Information

### CODE REVIEW

File: packaging/installer/netdata-updater.sh [hunk 1]
```diff
@@ -170,50 +167,46 @@ export NETDATA_LIB_DIR="${NETDATA_LIB_DIR:-${NETDATA_PREFIX}/var/lib/netdata}"
...
+  download "${NETDATA_TARBALL_CHECKSUM_URL}" "${TMPDIR}/sha256sum.txt"
+  download "${NETDATA_TARBALL_URL}" "${TMPDIR}/netdata-latest.gz.run"
+  if ! grep netdata-latest.gz.run "${TMPDIR}/sha256sum.txt" | safe_sha256sum -c - > /dev/null 2>&1; then
+    fatal "Static binary checksum validation failed. Stopping netdata installation and leaving binary in ${TMPDIR}"
+  fi
+
+  # Do not pass any options other than the accept, for now
+  sh "${TMPDIR}/netdata-latest.gz.run" --accept "${REINSTALL_OPTIONS}"
+
+  if ! rm -r "${TMPDIR}"; then
```
Comment (knatsakis) [inline packaging/installer/netdata-updater.sh]: Please see https://github.com/netdata/netdata/pull/7939#discussion_r373887175.

I realize now I should have commented here first.

Comment (prologic) [reply packaging/installer/netdata-updater.sh]: Fixed

File: packaging/installer/netdata-updater.sh [hunk 2]
```diff
@@ -170,50 +167,44 @@ export NETDATA_LIB_DIR="${NETDATA_LIB_DIR:-${NETDATA_PREFIX}/var/lib/netdata}"
...
+  cd "${TMPDIR}" || exit 1
+
+  download "${NETDATA_TARBALL_CHECKSUM_URL}" "${TMPDIR}/sha256sum.txt"
+  download "${NETDATA_TARBALL_URL}" "${TMPDIR}/netdata-latest.gz.run"
+  if ! grep netdata-latest.gz.run "${TMPDIR}/sha256sum.txt" | safe_sha256sum -c - > /dev/null 2>&1; then
+    fatal "Static binary checksum validation failed. Stopping netdata installation and leaving binary in ${TMPDIR}"
+  fi
+
+  # Do not pass any options other than the accept, for now
+  if ! sh "${TMPDIR}/netdata-latest.gz.run" --accept "${REINSTALL_OPTIONS}"; then
```
Comment (knatsakis) [inline packaging/installer/netdata-updater.sh]: But this way, you never run `rm -r "${TMPDIR}"`

```suggestion
  if sh "${TMPDIR}/netdata-latest.gz.run" --accept "${REINSTALL_OPTIONS}"; then
    rm -r "${TMPDIR}"
  else
    echo >&2 "NOTE: did not remove: ${TMPDIR}"
  fi
```

Comment (prologic) [reply packaging/installer/netdata-updater.sh]: Done and done

File: packaging/installer/netdata-updater.sh [hunk 3]
```diff
@@ -134,8 +133,7 @@ update() {
     pids=$(pidof netdata)
     do_not_start=
     if [ -n "${pids}" ]; then
-      #shellcheck disable=SC2086
-      kill -USR1 ${pids}
+      kill -USR1 "${pids}"
```
Comment (Ferroin) [inline packaging/installer/netdata-updater.sh]: This needs to be expanded. `pids` is a space separated list of PID's, which need to be passed as multiple arguments to `kill`.

File: packaging/installer/netdata-updater.sh [hunk 4]
```diff
@@ -202,16 +200,13 @@ if [ "${IS_NETDATA_STATIC_BINARY}" == "yes" ]; then
...
 
   # Do not pass any options other than the accept, for now
-  sh "${TMPDIR}/netdata-latest.gz.run" --accept ${REINSTALL_OPTIONS}
+  sh "${TMPDIR}/netdata-latest.gz.run" --accept "${REINSTALL_OPTIONS}"
 
-  #shellcheck disable=SC2181
-  if [ $? -eq 0 ]; then
-    rm -r "${TMPDIR}"
-  else
+  if ! rm -r "${TMPDIR}"; then
```
[the remark this answers is not in the document]
Comment (Ferroin) [reply packaging/installer/netdata-updater.sh]: Just noticed I wasn't seeing newer commits that fixed this, disregard.

File: packaging/installer/netdata-updater.sh [hunk 5]
```diff
@@ -202,16 +200,13 @@ if [ "${IS_NETDATA_STATIC_BINARY}" == "yes" ]; then
...
     echo >&2 "NOTE: did not remove: ${TMPDIR}"
   fi
   echo >&2 "Switching back to ${PREVDIR}"
-  cd "${PREVDIR}"
+  cd "${PREVDIR}" || exit
```
[the remark this answers is not in the document]
Comment (prologic) [reply packaging/installer/netdata-updater.sh]: GoodGoot catch! Fixed :)

File: packaging/installer/netdata-updater.sh (hunk 3 above)
Comment (prologic) [reply packaging/installer/netdata-updater.sh]: Good point. Fixed.

**Pull request merged**